arlo_server/__init__.py

```diff
@@ -867,8 +867,8 @@ def ballot_list(election_id, jurisdiction_id, round_id):
     query = (
         SampledBallotDraw.query.join(SampledBallot)
         .join(SampledBallotDraw.batch)
-        .join(AuditBoard)
         .join(Round)
+        .join(AuditBoard, AuditBoard.id == SampledBallot.audit_board_id)
```

Comment:
the explicit join clause here is so SQLAlchemy doesn't try to be too smart and automatically join on `round_id`, which then makes the existing flow break.